lib/vector-core/src/event/proto.rs

```diff
@@ -132,6 +132,12 @@ impl From<Metric> for event::Metric {
             .timestamp
             .map(|ts| chrono::Utc.timestamp(ts.seconds, ts.nanos as u32));
 
+        let interval_ms = if metric.interval_ms == 0 {
+            None
+        } else {
+            Some(metric.interval_ms)
+        };
```

Comment:
If you like `.then`:
```suggestion
        let interval_ms = (metric.interval_ms != 0).then(|| metric.interval_ms);
```